lib/file.js

```diff
@@ -3,33 +3,42 @@
 const debug = require('debug')('lint-staged:file')
 const fs = require('fs')
 
+const fsPromises = fs.promises
+
 /**
  * @param {String} filename
  * @returns {Promise<Buffer|Null>}
  */
-module.exports.readBufferFromFile = (filename, rejectENOENT = false) =>
-  new Promise(resolve => {
-    debug('Reading buffer from file `%s`', filename)
-    fs.readFile(filename, (error, buffer) => {
-      if (!rejectENOENT && error && error.code === 'ENOENT') {
-        debug("File `%s` doesn't exist, ignoring...", filename)
-        return resolve(null) // no-op file doesn't exist
-      }
-      debug('Done reading buffer from file `%s`!', filename)
-      resolve(buffer)
-    })
-  })
+const readFile = async filename => {
+  debug('Reading file `%s`', filename)
+  try {
+    return await fsPromises.readFile(filename)
+  } catch {
+    return null // no-op file doesn't exist
+  }
+}
 
 /**
  * @param {String} filename
  * @param {Buffer} buffer
  * @returns {Promise<Void>}
  */
-module.exports.writeBufferToFile = (filename, buffer) =>
-  new Promise(resolve => {
-    debug('Writing buffer to file `%s`', filename)
-    fs.writeFile(filename, buffer, () => {
-      debug('Done writing buffer to file `%s`!', filename)
-      resolve()
-    })
-  })
+const writeFile = async (filename, buffer) => {
+  debug('Writing file `%s`', filename)
+  await fsPromises.writeFile(filename, buffer)
+}
+
+/**
+ * @param {String} filename
+ * @returns {Promise<Void>}
+ */
+const removeFile = async filename => {
+  debug('Removing file `%s`', filename)
+  try {
+    await fsPromises.unlink(filename)
+  } catch {} // eslint-disable-line no-empty
```

Comment:
Same: why swallowing errors? In case something goes wrong it will be impossible to debug

Reply:
I got a bit lazy with the tests, since these only handle a narrow case. I’ll add some tests.